### PR TITLE
WIP: Latest post block iteration

### LINKED ISSUE #1594: Iterations on "Latest Posts" Block
_This description has been edited — please see issue history for older mockups!_

## Next steps

- Featured image support, in its own sidebar section.
	- [ ] Include core image size controls.
	- [ ] Include core alignment controls.
- Post content support, in its own sidebar section.
	- [x] Toggle on/off post content.
	- [x] When post content is toggled on, choose between excerpt or full post. (Am I missing a default core option? If it's just the two, maybe this should be a radio button instead).
	- [x] Slider to control excerpt length when the excerpt is selected.
- Post meta support, in its own sidebar section.
	- [x] Move "Display post date" into here. #13698 
	- [x] Add toggle to show/hide post author. #13698
- Organize settings
	- [x] Move "Order by" and "Category" into its own sidebar section, "Sorting and Filtering." #13698 
	- [x] Group "Number of posts" and "Columns" (when grid is selected) into their own sidebar section, "Display Settings." #13698 

## Why

In preparation for working on page templates in Gutenberg, we'll want a robust set of dynamic blocks that can be dropped into any post or page. Expanding this block will put us in a better position to tackle more complex dynamic or global blocks in the future.

Users shouldn't have to know how to write custom queries or understand the loop to add some posts to their homepage. The Recent Posts block is a great start, but to be a fully functioning solution, it needs to support more than titles and post dates.

### Before

image

image

### After

recent posts list all

recent posts grid all

- Added support for featured images, post content and excerpts, and post meta.
- Reorganized the sidebar settings into groups.
- Used the new Toolbar grouping design (unsure if this has been implemented yet).
- Updated the icons, but this is optional, we can continue using the existing Dashicons for now.

[View the complete prototype](https://wp.invisionapp.com/share/BNPZUYGYMZ8#/341120336_Recent_Posts-_List).

## Future iterations

h/t @paaljoachim for a couple of these suggestions:

- Expand on taxonomy offerings, like tags.
- CPT support.
- A way to update the "read more" text.
- A way to rearrange all the elements inside of the block — move up meta, move featured image below title, etc.
- Potentially display comment information (like # of comments).

## Note to new contributors

This issue itself is big but can be divided into smaller tasks which should be a great way to start contributing to Gutenberg with code. If you want to help please leave a comment and we will discuss what would be the best next step.

### PR DESCRIPTION
**Pull request by ajitbohra**
Fixes #1594 

Work In Progress

### CODE REVIEW

File: packages/block-library/src/latest-posts/edit.js [hunk 1]
```diff
@@ -185,12 +197,26 @@ class LatestPostsEdit extends Component {
...
 							<a href={ post.link } target="_blank">{ decodeEntities( post.title.rendered.trim() ) || __( '(Untitled)' ) }</a>
 							{ displayPostDate && post.date_gmt &&
 								<time dateTime={ format( 'c', post.date_gmt ) } className="wp-block-latest-posts__post-date">
 									{ dateI18n( dateFormat, post.date_gmt ) }
 								</time>
 							}
+
+							{ displayPostAuthor &&
+								<span className="wp-block-latest-posts__post-author">
+									{ __( 'by' ) } { post._embedded.author[ 0 ].name }
```
Comment (swissspidy) [inline packages/block-library/src/latest-posts/edit.js]: Should use `sprintf` here and add a translator comment.

File: packages/block-library/src/latest-posts/index.php [hunk 2]
```diff
@@ -49,6 +56,14 @@ function render_block_core_latest_posts( $attributes ) {
 			);
 		}
 
+		if( isset( $attributes['displayPostAuthor'] ) && $attributes['displayPostAuthor']) {
+			$list_items_markup .= sprintf(
+				'<span class="wp-block-latest-posts__post-author">%1$s %2$s</span>',
+				__( 'by' ),
```
Comment (swissspidy) [inline packages/block-library/src/latest-posts/index.php]: Same as in the JS, `sprintf` is needed here.

**Comment by getdave** [comment]
@ajitbohra I've left feedback here which you might find useful https://github.com/WordPress/gutenberg/pull/14627#issuecomment-479410698

**Comment by ajitbohra** [comment]
Closing this PR as most of the work here overlaps with #14627 which is already merged.

**Pull request closed without merging**